### ISSUE TITLE
kotlin.native.concurrent.InvalidMutabilityException with 1.3.3-native-mt

### ISSUE THREAD

**Issue by yigit** [ktorio/ktor#1543]
KTOR native with the new multi threaded coroutines fails with InvalidMutabilityException.
coroutines version: 1.3.3-native-mt
ktor version: 1.3.0-rc2
Here is the related code:

```
suspend fun doFetch() {
    HttpClient().use {
        val htmlContent = it.get<String>("https://en.wikipedia.org/wiki/Main_Page")
        println(htmlContent)
    }
}

fun doesNotWork() {
    runBlocking(Dispatchers.Default) { // remove Dispatchers.Default and it works
        doFetch()
    }
}

fun main() {
    doesNotWork()
}
```

to reproduce, checkout the following project:
https://github.com/yigit/ktor-coroutines-bug

run `./gradlew runDELX --stacktrace` from command line.

It fails with :

```
Uncaught Kotlin exception: kotlin.native.concurrent.InvalidMutabilityException: mutation attempt of frozen io.ktor.utils.io.ByteChannelNative@c006ba18
        at kfun:kotlin.Throwable.<init>(kotlin.String?)kotlin.Throwable (0x2d9607)
        at kfun:kotlin.Exception.<init>(kotlin.String?)kotlin.Exception (0x2d3115)
        at kfun:kotlin.RuntimeException.<init>(kotlin.String?)kotlin.RuntimeException (0x2d2cd5)
        at kfun:kotlin.native.concurrent.InvalidMutabilityException.<init>(kotlin.String)kotlin.native.concurrent.InvalidMutabilityException (0x2fff65)
        at ThrowInvalidMutabilityException (0x301673)
        at MutationCheck (0x5a742e)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.<set-waitingForRead>#internal (0x403fed)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.$awaitSuspendCOROUTINE$73.invokeSuspend(kotlin.Result<kotlin.Any?>)kotlin.Any? (0x417d9f)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.awaitSuspend(kotlin.Int)ValueType (0x418244)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.$readRemainingSuspendCOROUTINE$60.invokeSuspend#internal (0x40f22c)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.readRemainingSuspend#internal (0x40f5c0)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.$readRemainingCOROUTINE$59.invokeSuspend(kotlin.Result<kotlin.Any?>)kotlin.Any? (0x40e936)
        at kfun:io.ktor.utils.io.ByteChannelSequentialBase.readRemaining(kotlin.Long;kotlin.Int)io.ktor.utils.io.core.ByteReadPacket (0x40ec6e)
        at kfun:io.ktor.utils.io.readRemaining@io.ktor.utils.io.ByteReadChannel.()io.ktor.utils.io.core.ByteReadPacket (0x421354)
        at kfun:io.ktor.client.features.HttpPlainText.Feature.$install$lambda-1COROUTINE$26.invokeSuspend#internal (0x53c9c4)
        at kfun:io.ktor.client.features.HttpPlainText.Feature.$install$lambda-1COROUTINE$26.invoke#internal (0x53d3ef)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal (0x4d3e3d)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.proceed#internal (0x4d34be)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.proceedWith#internal (0x4d3697)
        at kfun:io.ktor.client.features.HttpCallValidator.Companion.$install$lambda-1COROUTINE$22.invokeSuspend#internal (0x538612)
        at kfun:io.ktor.client.features.HttpCallValidator.Companion.$install$lambda-1COROUTINE$22.invoke#internal (0x538baf)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal (0x4d3e3d)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.proceed#internal (0x4d34be)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.execute#internal (0x4d38c1)
        at kfun:io.ktor.util.pipeline.Pipeline.execute(#GENERIC_kotlin.Any;#GENERIC_kotlin.Any)#GENERIC_kotlin.Any (0x4ce690)
        at kfun:io.ktor.client.call.HttpClientCall.$receiveCOROUTINE$12.invokeSuspend(kotlin.Result<kotlin.Any?>)kotlin.Any? (0x5205d3)
        at kfun:io.ktor.client.call.HttpClientCall.receive(io.ktor.client.call.TypeInfo)kotlin.Any (0x520c06)
        at kfun:$doFetchCOROUTINE$0.invokeSuspend(kotlin.Result<kotlin.Any?>)kotlin.Any? (0x565423)
        at kfun:kotlin.coroutines.native.internal.BaseContinuationImpl.resumeWith(kotlin.Result<kotlin.Any?>) (0x2f4968)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal (0x4d4508)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal (0x4d4072)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal (0x4d5d9c)
        at kfun:kotlin.coroutines.native.internal.BaseContinuationImpl.resumeWith(kotlin.Result<kotlin.Any?>) (0x2f4c49)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal (0x4d4508)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal (0x4d4072)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal (0x4d5d9c)
        at kfun:kotlin.coroutines.native.internal.BaseContinuationImpl.resumeWith(kotlin.Result<kotlin.Any?>) (0x2f4c49)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal (0x4d4508)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal (0x4d4072)
        at kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal (0x4d5d9c)
        at kfun:kotlin.coroutines.native.internal.BaseContinuationImpl.resumeWith(kotlin.Result<kotlin.Any?>) (0x2f4c49)
        at kfun:kotlinx.coroutines.DispatchedTask.run() (0x3e2fca)
        at kfun:kotlinx.coroutines.EventLoopImplBase.processNextEvent()ValueType (0x3bb507)
        at kfun:kotlinx.coroutines.runEventLoop$kotlinx-coroutines-core(kotlinx.coroutines.EventLoop?;kotlin.Function0<kotlin.Boolean>) (0x3f665c)
        at kfun:kotlinx.coroutines.WorkerCoroutineDispatcherImpl.start$lambda-0#internal (0x3fe271)
        at kfun:kotlinx.coroutines.WorkerCoroutineDispatcherImpl.$start$lambda-0$FUNCTION_REFERENCE$80.invoke#internal (0x3fe3db)
        at kfun:kotlinx.coroutines.WorkerCoroutineDispatcherImpl.$start$lambda-0$FUNCTION_REFERENCE$80.$<bridge-UNN>invoke()#internal (0x3fe43b)
        at WorkerLaunchpad (0x30129c)
        at _ZN6Worker19processQueueElementEb (0x5ac1c9)
        at _ZN12_GLOBAL__N_113workerRoutineEPv (0x5aca2d)
        at  (0x7f54c8681c73)
        at clone (0x7f54c83b3def)
        at  ((nil))


```

**Comment by audkar** [comment]
Duplicate: https://github.com/ktorio/ktor/issues/1538

**Comment by petersamokhin** [comment]
@e5l any updates?

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-497">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed in 1.4.1
